pkg/controller/attach_detach.go

```diff
@@ -153,28 +153,29 @@ func (r *ReconcileAttachDetach) triggerAttach(ctx context.Context, azVolumeAttac
 
 	// initiate goroutine to attach volume
 	go func() {
-		response, attachErr := r.attachVolume(ctx, azVolumeAttachment.Spec.VolumeID, azVolumeAttachment.Spec.NodeName, azVolumeAttachment.Spec.VolumeContext)
+		goCtx := context.Background()
+		response, attachErr := r.attachVolume(goCtx, azVolumeAttachment.Spec.VolumeID, azVolumeAttachment.Spec.NodeName, azVolumeAttachment.Spec.VolumeContext)
```

Comment:
I believe we will want to have separate contexts for the `CloudProvisioner` operation (e.g. `attachVolume` in this case) and the API server operations (i.e. `UpdateCRIWithRetry`), at least for the future. It would probably be ok to pass `context.Background()` to the API server operations, but I wonder if we should use a conservative timeout value for the `CloudProvisioner` operation, e.g. 5 or 10 minutes just in case the request gets lost in the weeds?